src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -515,14 +515,11 @@ else if (listData instanceof Iterable<?> iterable) // fallback for the worst cas
 
 						// we store file generation conditions for current mod element
 						List<Boolean> conditionChecks = new ArrayList<>();
-						for (int i = 0; i < elements.size(); i++) {
-							conditionChecks.add(i,
-									!generatorTemplate.shouldBeSkippedBasedOnCondition(this, elements.get(i)));
+						for (int i = 0; i < items.size(); i++) {
+							conditionChecks.add(i, 
+									!generatorTemplate.shouldBeSkippedBasedOnCondition(this, items.get(i)));
 						}
-
-						// only add templates list if at least one list template will be generated
-						// or if we need all template lists possible for the given mod element (performFSTasks is false)
-						if (!conditionChecks.contains(true) && performFSTasks)
+						if (!conditionChecks.contains(true))

```

Comment:
This comment should be back, just modified to fit change.

Also why is performFSTasks now gone here?

Reply:
The way I wrote this with `performFSTasks` used, these template could compete for the same file even if they were not about to be generated.